ucp/_libs/ucp_py.pyx

```diff
@@ -729,7 +729,18 @@ def get_endpoint(peer_ip, peer_port):
     global reader_added
 
     ep = Endpoint()
-    ep.connect(peer_ip, peer_port)
+    deadline = time.time() + timeout
+    connection_established = False
+    while time.time() <= deadline:
+        if -1 == ep.connect(peer_ip, peer_port):
+            await asyncio.sleep(1)
```

Comment:
How expensive is it to try to connect?  If it's cheap then we might consider decreasing the sleep time here.